### ISSUE TITLE
[REQUEST]: Show selective compilers on wesbite

### ISSUE THREAD

**Issue by nnatu** [compiler-explorer/compiler-explorer#4447]
### Is your feature request related to a problem? Please describe

Currently a long list of compilers are showcased on website. Its time consuming to scroll through these and sometime you might missing what you are looking for. Is it possible to reduce this list

### Describe the solution you'd like

Can we have a selective compilers to reduce drop down list and enhance readability. Users can search for something they cant find in the drop . For eg currently under C language ARM 32-bit GCC we can have ARM gcc 1trunk , ARM gcc 12.2 and ARM gcc 12.1 and rest would be hidden but if user searches ARM gcc 11.3 it would displayed

### Describe alternatives you've considered

Cant think of any alternative solution

### Additional context

Not applicable

**Comment by partouf** [comment]
You can already do a lot of these searches, the only thing you can't currently do is to search for a specifically 32bits, but you can type in arm 1 to see all the possible arm compilers that have a 1 in the version

**Comment by RubenRBS** [comment]
Hi! Thanks for your request. We already implement a few options to help you find your compiler, but the fact that you have taken the time to open the request probably raises a good point about UX, we can surely work on it.

As @partouf mentions, you can search for the compiler you want, like this:
![imagen](https://user-images.githubusercontent.com/5364255/208190695-e41b128e-1458-43a0-a6e5-cd9b2968d1ae.png)

Another option is to click the star next to every compiler, that will mark it as favourite and show it at the top of the dropdown, which can also help you finding your most used compilers.

Hope some of it helps, and we'll keep in mind that we have lots still to do in this area :)

**Comment by nnatu** [comment]
@RubenRBS what does the .hidden option in config file do. Does it hide it from drop down but if I search I can find it?

**Comment by RubenRBS** [comment]
No, it completely hides it unless a link asks for that compiler id

**Comment by nnatu** [comment]
So there is no option to hide a compiler from drop down menu but make it available from search bar

**Comment by mattgodbolt** [comment]
> So there is no option to hide a compiler from drop down menu but make it available from search bar

Correct

**Comment by jeremy-rifkin** [comment]
#4849 should help with this. One option to address the concerns here would be to make a special group at the very top of the dropdown with the most recent versions of gcc, clang, and msvc for x86-64 for quick access. Maybe aarch64 or riscv too, but that's where the design decisions can get tricky.
